### PR TITLE
Use JDK7+ Objects.hash() and equals() rather than Guava

### PR DESCRIPTION
**Pull request by msgilligan**
(Guava says their methods are “deprecated” for JDK7+)

### CODE REVIEW

**Comment by msgilligan** [comment]
Not sure why that test failed. It passes on my Mac with OpenJDK 11.0.2.

**Comment by schildbach** [comment]
Don't worry about the test. I re-ran it and it passed.

File: core/src/main/java/org/bitcoinj/core/ECKey.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
-import com.google.common.base.Objects;
+import java.util.Objects;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/ECKey.java]: Import order isn't right here (multiple occurences).

File: core/src/main/java/org/bitcoinj/core/FilteredBlock.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 package org.bitcoinj.core;
 
-import com.google.common.base.Objects;
+import java.util.Objects;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/FilteredBlock.java]: Duplicate import.

Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/FilteredBlock.java]: Good catch. I'll fix it

File: core/src/main/java/org/bitcoinj/core/ECKey.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/ECKey.java]: Good catch. I'll fix it

**Comment by msgilligan** [comment]
OK, I force-pushed an update with correctly sorted `import`s

**Comment by schildbach** [comment]
Thanks! Merged.

**Pull request closed without merging**